### PR TITLE
Fixed unit test for accounts - Closes #709

### LINKED ISSUE #709: Fix unit test for accounts
In an effort to have more deterministic PRs,  we need to isolate this commit https://github.com/LiskHQ/lisk/commit/c2e73c6fd492e5d5d86a4b302902e62b4ef63c69 in a new PR.

### PR DESCRIPTION
**Pull request by diego-G**
Closes #709

### CODE REVIEW

File: test/unit/logic/account.js [hunk 1]
```diff
@@ -251,18 +265,37 @@ describe('account', function () {
 			});
 		});
 
-		it('should fetch results with limit of 50', function (done) {
-			account.getAll({limit: 50}, function (err, res) {
+		it('should fetch results with limit of 49', function (done) {
```
Comment (Isabello) [inline test/unit/logic/account.js]: limit still says 50??

File: test/unit/logic/account.js [hunk 2]
```diff
@@ -176,21 +176,35 @@ describe('account', function () {
 		});
 
 		it('should ignore limit if its < 1', function (done) {
```
Comment (4miners) [inline test/unit/logic/account.js]: `if below 1`

File: test/unit/logic/account.js [hunk 3]
```diff
@@ -176,21 +176,35 @@ describe('account', function () {
...
+			var sortedUsernames = _.sortBy(allAccounts, 'username').map(function (v) {
+				return {
```
Comment (4miners) [inline test/unit/logic/account.js]: make that 1 line

File: test/unit/logic/account.js [hunk 4]
```diff
@@ -176,21 +176,35 @@ describe('account', function () {
...
+				sort: {username: 1}
+			}, ['username'], function (err, res) {
 				expect(err).to.not.exist;
-				expect(res).to.eql(allAccounts);
+				expect(res).to.eql(sortedUsernames);
 				done();
 			});
 		});
 
 		it('should ignore offset if its < 1', function (done) {
```
Comment (4miners) [inline test/unit/logic/account.js]: same as above

File: test/unit/modules/accounts.js [hunk 5]
```diff
@@ -664,34 +428,34 @@ describe('account', function () {
 			});
 
 			it('should return top 10 accounts with respect to highest balance', function (done) {
```
Comment (4miners) [inline test/unit/modules/accounts.js]: `ordered by balance desc`

File: test/unit/modules/accounts.js [hunk 6]
```diff
@@ -663,35 +426,35 @@ describe('account', function () {
 				});
 			});
 
-			it('should return top 10 accounts with respect to highest balance', function (done) {
-				var topAccountsBalance = _.orderBy(allAccounts, 'balance').map(function (a) {
-					return a.balance;
-				}).reverse();
-
+			it('should return top 10 accounts ordered by balance desc', function (done) {
```
Comment (karmacoma) [inline test/unit/modules/accounts.js]: `should return top 10 accounts ordered by descending balance`

File: test/unit/logic/account.js [hunk 7]
```diff
@@ -94,7 +94,7 @@ describe('account', function () {
 			}).to.throw('Invalid public key, must be 64 characters long');
 		});
 
-		it.skip('should throw if parameter is not a hex string', function () {
+		it('should throw if parameter is not a hex string', function () {
```
Comment (karmacoma) [inline test/unit/logic/account.js]: `should throw error if parameter is not a hex string`

File: test/unit/modules/accounts.js (hunk 6 above)
Comment (Isabello) [inline test/unit/modules/accounts.js]: ` 		it('should sort the result according to field type in DESC', function (done) {`

From another area of code. Which format is right here?

Comment (SargeKhan) [inline test/unit/modules/accounts.js]: @karmacoma, which format would you suggest? Should I use `by descending balance` for specifying sort order & type throughout the file? Or keep it like 'in DESC'?

File: test/unit/modules/accounts.js [hunk 8]
```diff
@@ -663,35 +436,36 @@ describe('account', function () {
...
+				var limit = 10;
 
 				account.internal.top({
-					limit: 10
+					limit: limit
 				}, function (err, res) {
 					expect(err).to.not.exist;
 					expect(res.accounts).to.have.length(10);
-					expect(res.accounts.map(function (v) { return v.balance; })).to.equal(topAccountsBalance.slice(0, 10));
+					for (var i = 1; i < limit; i++) {
```
Comment (Isabello) [inline test/unit/modules/accounts.js]: for loops should start at 0 for validating arrays

File: test/unit/modules/accounts.js [hunk 9]
```diff
@@ -663,35 +436,36 @@ describe('account', function () {
...
-					offset: 10
+					limit: limit,
+					offset: offset
 				}, function (err, res) {
 					expect(err).to.not.exist;
 					expect(res.accounts).to.have.length(10);
-					console.log(res.accounts.map(function (v) { return v.balance; }));
-					console.log(topAccountsBalance.slice(10, 20));
-					expect(res.accounts.map(function (v) { return v.balance; })).to.eql(topAccountsBalance.slice(10, 20));
+					for (var i = 1; i < limit; i++) {
```
Comment (Isabello) [inline test/unit/modules/accounts.js]: Same as previous

File: test/unit/modules/accounts.js [hunk 10]
```diff
@@ -663,35 +436,36 @@ describe('account', function () {
...
+					offset: offset
 				}, function (err, res) {
 					expect(err).to.not.exist;
 					expect(res.accounts).to.have.length(10);
-					console.log(res.accounts.map(function (v) { return v.balance; }));
-					console.log(topAccountsBalance.slice(10, 20));
-					expect(res.accounts.map(function (v) { return v.balance; })).to.eql(topAccountsBalance.slice(10, 20));
+					for (var i = 0; i < limit - 1; i++) {
+						var isLarger = (new bignum(res.accounts[i].balance)).gte(new bignum(res.accounts[i + 1].balance));
+						expect(isLarger).to.equal(true);
```
Comment (MaciejBaj) [inline test/unit/modules/accounts.js]: `expect(inew bignum(res.accounts[i].balance)).gte(new bignum(res.accounts[i + 1].balance));`

File: test/unit/modules/accounts.js [hunk 11]
```diff
@@ -663,35 +436,36 @@ describe('account', function () {
...
 				account.internal.top({
-					limit: 10
+					limit: limit
 				}, function (err, res) {
 					expect(err).to.not.exist;
 					expect(res.accounts).to.have.length(10);
-					expect(res.accounts.map(function (v) { return v.balance; })).to.equal(topAccountsBalance.slice(0, 10));
+					for (var i = 0; i < limit - 1; i++) {
+						var isLarger = (new bignum(res.accounts[i].balance)).gte(new bignum(res.accounts[i + 1].balance));
+						expect(isLarger).to.equal(true);
```
Comment (MaciejBaj) [inline test/unit/modules/accounts.js]: merge into 1 as shown below

**Pull request merged**